cpp/open3d/t/pipelines/registration/Registration.cpp

```diff
@@ -95,33 +96,32 @@ RegistrationResult EvaluateRegistration(const geometry::PointCloud &source,
     geometry::PointCloud source_transformed = source.Clone();
     source_transformed.Transform(transformation);
 
-    open3d::core::nns::NearestNeighborSearch target_nns(
-            target.GetPointPositions());
+    core::nns::NearestNeighborSearch target_nns(target.GetPointPositions());
 
     bool check = target_nns.HybridIndex(max_correspondence_distance);
     if (!check) {
         utility::LogError(
                 "NearestNeighborSearch::HybridSearch: Index is not set.");
     }
 
-    RegistrationResult result(transformation, false);
-    GetRegistrationResultAndCorrespondences(source_transformed, target_nns,
-                                            max_correspondence_distance,
-                                            transformation, result);
-    return result;
+    return GetRegistrationResultAndCorrespondences(
+            source_transformed, target_nns, max_correspondence_distance,
+            transformation);
 }
 
-RegistrationResult ICP(const geometry::PointCloud &source,
-                       const geometry::PointCloud &target,
-                       const double max_correspondence_distance,
-                       const core::Tensor &init_source_to_target,
-                       const TransformationEstimation &estimation,
-                       const ICPConvergenceCriteria &criteria,
-                       const double voxel_size,
-                       const bool save_loss_log) {
+RegistrationResult ICP(
+        const geometry::PointCloud &source,
+        const geometry::PointCloud &target,
+        const double max_correspondence_distance,
+        const core::Tensor &init_source_to_target,
+        const TransformationEstimation &estimation,
+        const ICPConvergenceCriteria &criteria,
+        const double voxel_size,
+        std::function<void(std::unordered_map<std::string, core::Tensor> &)>
```

Comment:
We should be able to pass functions with `const &`. Could you check and change all of them?